yarn-project/end-to-end/Earthfile

```diff
@@ -14,8 +16,8 @@ E2E_TEST_LOCAL:
   ENV TEST=$test
   ENV DEBUG=$debug
   WITH DOCKER \
-    --load aztecprotocol/aztec:latest=../+aztec \
-    --load aztecprotocol/end-to-end:latest=../+end-to-end \
+    --load aztecprotocol/aztec:latest=(../+aztec --use_local_builds=$use_local_builds) \
```

Comment:
Can we push the global arg all the way?

Reply:
Couldn't get it working, I had to explicitly pass it from here, though it's possible I was doing something wrong. Based on [their docs](https://docs.earthly.dev/docs/guides/build-args):

> Arg overrides within the same Earthfile are passed automatically to each other. This behavior does not apply to references to other Earthfiles. In order to pass arguments to other Earthfiles, you must either explicitly pass the argument. 

There's a `--pass-args` option you can use with `FROM` commands, but couldn't make it work here.

Overall I've found Earthly args to be a bit confusing: I'd see a value displayed in the logs, but then the commands that used them couldn't actually "see" them unless I explicitly declare the `ARG` in that function or pass it.

Reply:
Right the `ARG`'s having to be explicitly declared seems a fair design decision to me. But the fact that it only works within a single Earthfile I do find weird